### PR TITLE
EcalPedestals management

### PR DESCRIPTION
**Pull request by depasse**
EcalPedestals tools  + management (by splitting) of tags with thousands IOVs + old DBv1 pythons upgraded

### CODE REVIEW

**Comment by franzoni** [comment]
please test

**Comment by ggovi** [comment]
@depasse 
Pierre could you please remove from the C++ all of the printouts with the DB user/password.
From the python code, you should remove the hard-coded DB credentials and go for a .netrc file solution ( see for example https://github.com/cms-sw/cmssw/blob/CMSSW_9_2_X/CondTools/Ecal/python/EcalDAQ_popcon.py )

**Comment by davidlange6** [comment]
and change the password(s) in the PR...

**Comment by depasse** [comment]
Could you give me the syntax to put the right info in the .netrc  ?

When I test the modified python : I obtain :
Exception Message:
python encountered the error: <type 'exceptions.Exception'>
Entry for service cms_omds_adg/CMS_ECAL_R not found in .netrc

Actually, I only have this line in .netrc (for uploadConditions.py):

machine ConditionUploader login depasse password xxx

So I suppose login : CMS_ECAL_R password : the usual one  but what is the full line to insert ?

Thanks.

Pierre

________________________________

**Comment by ggovi** [comment]
@depasse 
the line is of the type:
machine <label> login <username> password <pwd>
depending of what you need to read, you set the label accordingly. See the example:

https://github.com/cms-sw/cmssw/blob/CMSSW_9_2_X/CondTools/Ecal/python/db_credentials.py

**Comment by depasse** [comment]
Hi,

so my .netrc   file looks like :

machine ConditionUploader login depasse password *****
machine cms_omds_adg/CMS_ECAL_R login cms_ecal_r password *****

and it works well.

Pierre
________________________________

**Comment by ggovi** [comment]
please test

**Comment by ggovi** [comment]
+1

File: CondTools/Ecal/src/EcalPedestalsHandler.cc [hunk 1]
```diff
@@ -648,16 +648,839 @@ void popcon::EcalPedestalsHandler::getNewObjectsH2()
...
+	   >> EBmean1[iChannel] >> EBrms1[iChannel];
+    if(hashedId != iChannel + 1) {
+      std::cout << m_filename << " strange hash " << hashedId << " while iChannel " << iChannel << std::endl;
+      exit(-1);
+    }
+  }
+
+  // ***************** now EE *****************
+  float EEmean12[kEEChannels], EErms12[kEEChannels], EEmean6[kEEChannels], EErms6[kEEChannels], EEmean1[kEEChannels], EErms1[kEEChannels];
+  for (int iChannel = 0; iChannel < 14648; iChannel++) {
```
Comment (davidlange6) [inline CondTools/Ecal/src/EcalPedestalsHandler.cc]: hi @depasse - I gather 14648 can be instead kEEChannels?

**Comment by davidlange6** [comment]
otherwise you need to squash the commits with passwords (and assure me they are changed)

**Comment by depasse** [comment]
Hi,

in the files  :
copyFilePedestals.py, copyTreePedestals.py, EcalO2O_SRP_cfg.py, updateADCToGeV_test.py :

I have replaced         OnlineDBPassword = cms.string(****) by the structure suggested by Giacomo Govi :

import CondTools.Ecal.db_credentials as auth
db_service,db_user,db_pwd = auth.get_readOnly_db_credentials()
        OnlineDBUser = cms.string(db_user),
        OnlineDBPassword = cms.string(db_pwd),
        OnlineDBSID = cms.string(db_service),
...
to access this Read-Only DB.

________________________________

**Comment by davidlange6** [comment]
Yes, but it's all still in the commit history....


On Jun 2, 2017, at 6:09 PM, depasse <notifications@github.com<mailto:notifications@github.com>> wrote:

Hi,

in the files :
copyFilePedestals.py, copyTreePedestals.py, EcalO2O_SRP_cfg.py, updateADCToGeV_test.py :

I have replaced OnlineDBPassword = cms.string(****) by the structure suggested by Giacomo Govi :

import CondTools.Ecal.db_credentials as auth
db_service,db_user,db_pwd = auth.get_readOnly_db_credentials()
OnlineDBUser = cms.string(db_user),
OnlineDBPassword = cms.string(db_pwd),
OnlineDBSID = cms.string(db_service),
...
to access this Read-Only DB.

________________________________

**Comment by depasse** [comment]
What to do then ?

git rm ?
________________________________

**Comment by depasse** [comment]
So I must close this PR and create a new one ?
________________________________

**Comment by davidlange6** [comment]
that or use git merge --squash (where I don't have an authoritative recipe, but google has plenty of information)

**Comment by depasse** [comment]
Create a new PR to avoid password diffusion pb

**Pull request closed without merging**